test/test_relinker_functionality.py

```
@@ -0,0 +1,306 @@
+        
+        # Test absolute path
+        paths.append('lib3.a', '*', '/absolute/path/lib3.a')
+        result = paths.index('lib3.a', '*')
+        assert result[0] == '/absolute/path/lib3.a', "Absolute path should remain unchanged"
+        print('✓ Absolute path handling works')
+        
+        # Test missing IDF_PATH error
+        del os.environ['IDF_PATH']
+        try:
+            paths.append('lib4.a', '*', '$IDF_PATH/test.a')
+            assert False, "Should raise error for missing IDF_PATH"
```

Ruff (0.15.7)

[warning] Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)